Source/Core/DolphinQt/MainWindow.cpp

```diff
@@ -275,6 +278,14 @@ MainWindow::MainWindow(Core::System& system, std::unique_ptr<BootParameters> boo
   AchievementManager::GetInstance().Init(reinterpret_cast<void*>(winId()));
   if (AchievementManager::GetInstance().IsHardcoreModeActive())
     Settings::Instance().SetDebugModeEnabled(false);
+#ifdef RC_CLIENT_SUPPORTS_RAINTEGRATION
+  Config::AddConfigChangedCallback(
+      [this]() { QueueOnObject(this, [this] { this->OnHardcoreChanged(); }); });
+#endif  // RC_CLIENT_SUPPORTS_RAINTEGRATION
```

Comment:
This shouldn't be gated behind RC_CLIENT_SUPPORTS_RAINTEGRATION.

Reply:
Also, you mentioned on Discord that this needs to react both to the hardcore setting changing and to the RetroAchievements enabled setting changing. This is non-obvious, so I would recommend adding a comment about it.